### ISSUE TITLE
no mono legato in NoteLoop

### ISSUE THREAD

**Issue by MaZe6K** [MarcBoule/ImpromptuModular#78]
Hi there!

It seems that NoteLoop doesn't allow mono legato and always keeps the first pitch received with a continuous gate.

Would be nice for mono keyboard playing to work with legato.

Other than that I'm very happy with it.

**Comment by MarcBoule** [comment]
Hi, yes, this is a known limitation unfortunately (which I mention in the manual), and also the reason why the module has the "Sample and hold" tag. It is a builtin limitation given that the delay engine is not continually sampling the CV input like in normal audio-based delays. This keeps the memory requirements very low for the module, but does have that limitation. I'll keep it in mind though, so thanks for mentioning it! Cheers!

**Comment by MaZe6K** [comment]
I think a nice solution would be to add a "mono legato output mode" - the recording could stay the same, and it could also work as a poly-to-mono cv/gate converter when not looping.

**Comment by MarcBoule** [comment]
Just to make sure I understand, it would merge all the poly channels of the gate output into one mono signal, and same for the CV? For the gate, all the poly channels would be effectively combined in a big logical OR gate (?), and for the CV, it would simply output the CV channel corresponding to the last poly gate that went high? I think that makes the most sense, since we can't really merge CV signals (i.e. adding them up would likely not be very useful), but if this is what you have in mind, I'll think about it more.

**Comment by MaZe6K** [comment]
Yes that is what I thought, cv output is always the last one, gate summed or ORed.

CV could be the last one that comes with a corresponding gate, or the last one to change, which would amount to the same output I think. As a bonus there could be modes for last, lowest or highest note.

My use case would be playing it with a keyboard, always with last mode, but other modes might be more useful for playing with chord generator modules and the like, where multiples keys arrive at exactly the same time.

**Comment by MaZe6K** [comment]
I asked here if there is a module that does this, and it seems Ahornberg is building one.

https://community.vcvrack.com/t/something-to-turn-a-poly-cv-gate-into-mono-legato/22867

**Comment by MarcBoule** [comment]
Ok, I'll think about this a bit more, but I'll open the issue again while I do so. If Ahornberg is making a module for this kind of merging, it would be ideal, since it could then be used in other contexts also (the advantages of modular, haha!).

**Comment by MarcBoule** [comment]
are you on windows? I've coded an initial version for you to test

**Comment by MaZe6K** [comment]
Oh nice! Yes, win 10. If you have binaries I'll check it out.

**Comment by MarcBoule** [comment]
[ImpromptuModular-2.5.0-win-x64.vcvplugin.zip](https://github.com/user-attachments/files/17129721/ImpromptuModular-2.5.0-win-x64.vcvplugin.zip)
Here it is, you just need to unzip it first.
Let me know if it's what you had in mind, and if it works well.

**Comment by MaZe6K** [comment]
Yes! Seems to work perfect! Thank you!

It's very nice to have this looper in Rack, very happy with it!

**Comment by MarcBoule** [comment]
Excellent! :-) It might take a while for this to be submitted to the library, but in the mean time at least you have this version.
Glad you like it, cheers!
